enhancements/update/eus-upgrades-mvp.md

```diff
@@ -0,0 +1,333 @@
+failure domain because anti-affinity scheduling is used in most scenarios. We
+should provide some mechanism to rank upgraded nodes as slightly higher priority
+in scheduler configuration. Perhaps a timestamp of configuration generation that
+the scheduler then biases toward nodes with the greatest timestamps as a last in
+chain priority.
+
+We should consider enabling the rescheduler as well to ensure that when the
+upgrade completes we rebalance so that the last node ends up with proportionate
+workload.
+
+#### As an admin of a disconnected cluster, I have a tool which helps me mirror all content necessary for EUS to EUS upgrades
+
+We should build a graph viewer / solver somewhere which allows an admin of a 4.6
+cluster hoping to upgrade to 4.10 to discover all content which must be mirrored
+between their current version and their eventual 4.10 target.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The main caveat of what's outlined here is that it does not reduce reboots
+which, for many reasons, is highly desired by some admins. The rationale for not
+addressing that in this enhancement is that this enhancement targets
+improvements which benefit all upgrades.
+
+The epic / user story to backport API removal notifications to 4.6 assumes that
+we will require anyone upgrading along an EUS specific upgrade path to update to
+a 4.6 version which is specific to EUS. This is necessary because only those who
+follow the EUS upgrade path have 4.10 as their next logical target version. We
+should not alert non EUS clusters of removals in 4.10 until they've upgraded to
+a later minor version.
+
+As with all upgrades, new features enabled by interim versions are not expected
+enabled during the upgrade path. This means that for instance new MachineConfig
+features enabling complex root devices while introduced in 4.7 may not be utilized
+until the cluster has completed its upgrade.
+
+### Risks and Mitigations
+
+The improvements here do not seem to introduce risk unto themselves. However
+failure to implement some of these such as the version skew constraints would
+impose significant risk to supportability in clusters which may upgrade more
+rapidly than is safe to do so.
+
+Additionally, the timeframe for final delivery extends three releases into the
+future. This allows for design iteration during the 4.8 release cycle with
+implementation focused on the 4.9 and 4.10 release cycles.
+
+## Design Details
+
+### Open Questions [optional]
+
+- As this is a bit of a meta-enhancement calling out epic level work it should be
+no surprise that there will be implementation details unaccounted for today. We
+will need to decide on an epic by epic level which require enhancements of their
+own.
+
+- While I believe all of this work to be prerequisite to taking on work to skip
+upgrade steps that may not be widely agreed upon.
+
+- Do we need to consider CCO for similar treatment that we're providing for API
```

Comment:
@dgoodwin @joelddiaz Your thoughts here please

Reply:
As long as we visit each minor release between 4.6->4.10, I think we're okay. We just need to make sure the minor release visited is a release where CCO knows about the next minor release requirements. So for example: you can't go from 4.6.10 (which can alert about upgradeability to 4.7) to 4.7.2 (which doesn't know about upgradeability to 4.8 b/c 4.7.2 was too early to know about what was coming in 4.8).

Reply:
We'd always be respecting minimum version requirements, we'll be upgrading along very late 4.6.z, 4.7.z, and 4.8.z. While 4.9.z will be relatively young it must meet the minimum version requirements to upgrade to 4.10.